spec/overcommit/hook_context/pre_commit_spec.rb

```diff
@@ -66,6 +66,30 @@
       end
     end
 
+    context 'when only a submodule change is staged' do
+      around do |example|
+        submodule = repo do
+          `git commit --allow-empty -m "Initial commit"`
+        end
+
+        repo do
+          `git submodule add #{submodule} sub &>/dev/null`
+          `git commit -m "Add submodule"`
+          `echo "Hello World" > sub/submodule-file`
+          `git submodule foreach 'git add submodule-file'`
+          `git submodule foreach 'git commit -m "Another commit"'`
+          `git add sub`
+          example.run
+        end
+      end
+
+      it 'keeps staged submodule change' do
+        subject
+        `git diff`.should be_empty
```

Comment:
Would it make more sense to check that the submodule change was committed, rather than checking that the diff was empty? It's possible that we simply lost all changes, in which case the diff would still be empty.